docs/RxSwift.md

```diff
@@ -12,6 +12,8 @@ Just use your `MoyaProvider` instance.
 let provider = MoyaProvider<GitHub>()
 ```
 
+*You are responsible to retain the provider for as long as you need. If you fail to do so, the object will be automatically disposed and the observable will not work properly.*
```

Comment:
Hm, this is more of a general rule for all of the configurations: `Moya`, `Moya/RxSwift`, `Moya/ReactiveSwift`. Can we move it somewhere to e.g. Basic usage? Or maybe we should duplicate it in all the docs to make sure everyone sees it? 🤔

Reply:
I agree. It would probably be good in `Providers.md`, `RxSwift.md`, `ReactiveSwift.md`. If you want to add basic too that's fine 😅

Reply:
Oh, now that I think about it duplicating it in 4 places might not be the best idea. Maybe just cover it in `Basic.md` and  `Providers.md`?

Reply:
Yep, I think `Basic.md` would be good. Pretty nice to at least have read when you start using Moya, as the Xcode errors might not lead you to a fix immediately.

Reply:
Yeah but this keeps coming up mainly on the reactive providers? Or is it just because most people use the reactive Moya libraries?

Reply:
I think the vast majority of people using Moya is using RxSwift extensions. At least it was the case few months back. When you start with Moya, whether your use reactive extensions or not, you should read the basics as well. I'm thinking if providers also should have it, though. 🤔

Reply:
Ok sorry to bikeshed the issue 😅 I think `Providers.md` and `Basic.md` will be clear enough

Reply:
Okay cool, I agree 👍

Reply:
Okay cool, I agree 👍 Sorry for the delay in this one, @MrAsterisco, we got an agreement now 😄

Reply:
Ok, so just to confirm: we need to move the very same sentence to both `Providers.md` and `Basic.md`, right?

Reply:
Not exactly the very same sentence, because we have `Observable` and `disposing` here. We would need to adopt it to a more broader use (if you don't retain the provider, it (probably) will be released before the response comes back). But yes, remove it from `RxSwift.md` and move to `Providers.md` and `Basic.md` 👍